### PR TITLE
Editable Session for Owner

### PR DESCRIPTION
**Pull request by TS-Daniel16**
Allows session owners to create annotations always independent of editable setting

Fixes different time handling between non-audio and audio sessions

Adds first version of authorization handling for subscriptions

Switches isAudioSession and isSessionOwner handling to server side and removes it from client side

### CODE REVIEW

**Comment by TS-Daniel16** [comment]
Solves #280

**Comment by TS-Daniel16** [comment]
@TS-Stefan @TS-Tino16  this is ready for review

File: apps/api/src/app/sessions/audio-session.interceptor.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import { Observable, map } from 'rxjs';
+import { Session } from './entities/session.entity';
+import { isArray } from 'class-validator';
+
+@Injectable()
+export class AudioSessionInterceptor implements NestInterceptor {
+  intercept(context: ExecutionContext, next: CallHandler): Observable<Session | Session[]> {
+    return next.handle().pipe(
+      map((data: any) => {
+        if (isArray(data)) {
```
Comment (TS-Tino16) [inline apps/api/src/app/sessions/audio-session.interceptor.ts:11]: is this the same as Array.isArray function?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray

**Review by TS-Tino16 — APPROVED** [review]
Looks good, just two remarks

File: apps/api/src/app/sessions/audio-session.interceptor.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+          data.map((session: Session) => {
+            session.isAudioSession = !!session.mediaId;
```
Comment (TS-Tino16) [inline apps/api/src/app/sessions/audio-session.interceptor.ts:13]: Do we need an interceptor here? Can't we just add this field as a getter to the session entity?

File: .gitignore [hunk 3]
```diff
@@ -20,7 +20,6 @@ node_modules
 # IDE - VSCode
 .vscode/*
 !.vscode/tasks.json
-!.vscode/launch.json
```
[the remark this answers is not in the document]
Comment (TS-Daniel16) [reply .gitignore:23]: Yes, we have not yet added a common launch.json and I didn't necessarily add my (very not common) launch json

File: apps/api/src/app/sessions/audio-session.interceptor.ts (hunk 1 above)
Comment (TS-Daniel16) [reply apps/api/src/app/sessions/audio-session.interceptor.ts:11]: Not sure what is used inside class-validator for the array check. In the end I used it under the assumption that we also trust class-validator with this in input objects/DTOs, so should be good enough here as well (so far it differentiated correctly)

File: apps/api/src/app/sessions/audio-session.interceptor.ts (hunk 2 above)
Comment (TS-Daniel16) [reply apps/api/src/app/sessions/audio-session.interceptor.ts:13]: Currently this is only used on UI level not on backend level. That's why I added it only in the layer where the session entity is requested (of course I could have added it in the resolver itself, but the interceptor felt like a better place to do something in addition for some of the requests - the subscription request should most likely be excluded in the future). Adding it in a lower layer would always calculate it and add it to the entity even for "internal" requests

**Pull request merged**